Sources/QuaternionModule/Quaternion.swift

```diff
@@ -0,0 +1,365 @@
+  @usableFromInline @inline(__always)
+  internal init(from components: SIMD4<RealType>) {
+    self.components = components
+  }
+}
+
+// MARK: - Basic Property
+extension Quaternion {
+  /// The real part of this quaternion.
+  ///
+  /// If `q` is not finite, `q.real` is `.nan`.
+  public var real: RealType {
+    @_transparent
+    get { isFinite ? components[0] : .nan }
+
+    @_transparent
+    set { components[0] = newValue }
+  }
+
+  /// The imaginary part of this quaternion.
+  ///
+  /// If `q` is not finite, `q.imaginary` is `.nan` in all lanes.
+  public var imaginary: SIMD3<RealType> {
+    @_transparent
+    get { isFinite ? components[SIMD3(1,2,3)] : SIMD3(repeating: .nan) }
+
+    @_transparent
+    set {
+      components = SIMD4(components[0], newValue.x, newValue.y, newValue.z)
+    }
+  }
+
+  /// The additive identity, with real and *all* imaginary parts zero.
+  ///
+  /// See also:
+  /// -
+  /// - .one
+  /// - .i
+  /// - .infinity
+  @_transparent
+  public static var zero: Quaternion {
+    Quaternion(from: SIMD4(repeating: 0))
+  }
+
+  /// The multiplicative identity, with real part one and *all* imaginary parts zero.
+  ///
+  /// See also:
+  /// -
+  /// - .zero
+  /// - .i
+  /// - .infinity
+  @_transparent
+  public static var one: Quaternion {
+    Quaternion(from: SIMD4(1,0,0,0))
+  }
+
+  /// The imaginary unit.
+  ///
+  /// See also:
+  /// -
+  /// - .zero
+  /// - .one
+  /// - .infinity
+  @_transparent
+  public static var i: Quaternion {
+    Quaternion(imaginary: SIMD3(repeating: 1))
+  }
+
+  /// The point at infinity.
+  ///
+  /// See also:
+  /// -
+  /// - .zero
+  /// - .one
+  /// - .i
+  @_transparent
+  public static var infinity: Quaternion {
+    Quaternion(.infinity)
+  }
+
+  /// The conjugate of this quaternion.
+  @_transparent
+  public var conjugate: Quaternion {
+    Quaternion(from: components.replacing(with: -components, where: [false, true, true, true]))
+  }
+
+  /// True if this value is finite.
+  ///
+  /// A quaternion is finite if neither component is an infinity or nan.
+  ///
+  /// See also:
+  /// -
+  /// - `.isNormal`
+  /// - `.isSubnormal`
+  /// - `.isZero`
+  /// - `.isPure`
+  @_transparent
+  public var isFinite: Bool {
+    return components.x.isFinite
+        && components.y.isFinite
+        && components.z.isFinite
+        && components.w.isFinite
+  }
+
+  /// True if this value is normal.
+  ///
+  /// A quaternion is normal if it is finite and *either* the real or *all* of the imaginary
+  /// components are normal. A floating-point number representing one of the components is normal
+  /// if its exponent allows a full-precision representation.
+  ///
+  /// See also:
+  /// -
+  /// - `.isFinite`
+  /// - `.isSubnormal`
+  /// - `.isZero`
+  /// - `.isPure`
+  @_transparent
+  public var isNormal: Bool {
+    let realIsNormal = components.x.isNormal
+    let imaginaryIsNormal = components.y.isNormal && components.z.isNormal && components.w.isNormal
```

Comment:
This should be normal if _any_of the components is normal, not all of them.

Reply:
My bad, thank you!